newrelic-agent/src/main/java/com/newrelic/agent/transport/DataSenderImpl.java

```diff
@@ -669,15 +680,19 @@ private void throwExceptionFromStatusCode(String method, ReadResult result, byte
                 }
             case HttpResponseCode.UNAUTHORIZED:
                 // received 401 Unauthorized, throw exception instead of parsing LicenseException from 200 response body
+                SuperAgentIntegrationUtils.reportUnhealthyStatus(healthDataChangeListeners, AgentHealth.Status.INVALID_LICENSE);
```

Comment:
Any concerns about having this stuff mixed into the Datasender if the super agent is disabled?

Reply:
Good idea. I'll refactor and wrap these calls around a config check gate just to be double safe